src/inject/dynamic-theme/watch.ts

```diff
@@ -179,9 +179,11 @@ export function watchForStyleChanges(update: (styles: ChangedStyles) => void) {
             return;
         }
         const shadowObserver = new MutationObserver(handleMutations);
-        shadowObserver.observe(node.shadowRoot, mutationObserverOptions);
-        shadowObservers.add(shadowObserver);
-        nodesShadowObservers.set(node, shadowObserver);
+        if (shadowObserver.takeRecords().length != 0) { // Don't handle empty Shadowobservers
+            shadowObserver.observe(node.shadowRoot, mutationObserverOptions);
```

Comment:
As far as I understand, the error happens when shadowRoot is not an instance of Shadow Root.